src/sqlancer/StatementReducer.java

```diff
@@ -28,32 +33,59 @@ public void reduce(G state, Reproducer<G> reproducer, G newGlobalState) throws E
         printQueries(knownToReproduceBugStatements);
         System.out.println();
 
-        do {
+        if (knownToReproduceBugStatements.size() <= 1) {
+            return;
+        }
+
+        partitionNum = 2;
+        while (knownToReproduceBugStatements.size() >= 2) {
```

Comment:
Maybe we can set a threshold so that only when `knownToReproduceBugStatements.size()` is very large do we use this method, and we use the normal full loop to handle cases when `knownToReproduceBugStatements.size()` is relatively small. 

JDK's implementation of the sort() algorithm is similar. If this sounds reasonable, we can discuss the threshold with @mrigger 


![image](https://user-images.githubusercontent.com/63448884/224475800-16158f6c-8f5f-4077-a920-8e43d0698f1c.png)